### PR TITLE
Volume reset for alsabat testing

### PR DESCRIPTION
**Pull request by keqiaozhang**

### CODE REVIEW

File: case-lib/lib.sh [hunk 1]
```diff
@@ -833,7 +833,7 @@ reset_sof_volume()
     amixer -Dhw:0 scontrols | sed -e "s/^.*'\(.*\)'.*/\1/" |egrep 'PGA|gain' |
     while read -r mixer_name
     do
-        amixer -Dhw:0 -- sset "$mixer_name" 0dB
```
Comment (singalsu) [inline case-lib/lib.sh:836]: @ujfalusi Do you have idea why 0 dB is not working?

File: case-lib/lib.sh [hunk 2]
```diff
@@ -827,10 +827,10 @@ set_alsa_settings()
     esac
 }
 
-reset_PGA_volume()
+reset_sof_volume()
 {
     # set all PGA* volume to 0dB
-    amixer -Dhw:0 scontrols | sed -e "s/^.*'\(.*\)'.*/\1/" |grep PGA |
+    amixer -Dhw:0 scontrols | sed -e "s/^.*'\(.*\)'.*/\1/" |egrep 'PGA|gain' |
```
Comment (marc-hb) [inline case-lib/lib.sh:833]:
```suggestion
    amixer -Dhw:0 scontrols | sed -e "s/^.*'\(.*\)'.*/\1/" | grep -E 'PGA|gain' |
```
see shellcheck

**Comment by marc-hb** [comment]
only suspend/resume timeouts in https://sof-ci.01.org/softestpr/PR1010/build181/devicetest/index.html but some alsabat failures in https://sof-ci.01.org/softestpr/PR1010/build182/devicetest/index.html

**Comment by keqiaozhang** [comment]
> but some alsabat failures in https://sof-ci.01.org/softestpr/PR1010/build182/devicetest/index.html

These failures will be fixed in https://github.com/thesofproject/sof-test/pull/992.

**Comment by keqiaozhang** [comment]
After more tests, I found not all gain controls can set to 0dB,
For example:
```
amixer -Dhw:0 -- sset 'gain.0.1 1 Playback Volume 0' 0dB
amixer: Invalid command!

amixer -Dhw:0 -- sset "gain.100.1 DMIC0 Capture Volume 1" 0dB
amixer: Invalid command!
```
This is still a problem that needs to be clarified.

**Comment by singalsu** [comment]
I'm wondering if there has ever been a working dB setting in amixer. 0dB has worked but it doesn't seem accept negative values for attenuation.

```
$ amixer sset 'PGA1.0 1 Master' 0dB
Simple mixer control 'PGA1.0 1 Master',0
  Capabilities: pvolume
  Playback channels: Front Left - Front Right
  Limits: Playback 0 - 32
  Mono:
  Front Left: Playback 32 [100%] [0.00dB]
  Front Right: Playback 32 [100%] [0.00dB]
$ amixer sset 'PGA1.0 1 Master' -6dB
amixer: invalid option -- '6'
Invalid switch or option -? needs an argument.
amixer: invalid option -- 'B'
Invalid switch or option -? needs an argument.

```
The manual pages suggests that it should be possible but maybe there's some bug

```
       set or sset <SCONTROL> <PARAMETER> ...
              Sets  the  simple  mixer control contents. The parameter can be the volume either as a percentage from 0% to
              100% with % suffix, a dB gain with dB suffix (like -12.5dB), or an exact hardware value.  The dB gain can be
              used only for the mixer elements with available dB information.  When plus(+) or minus(-) letter is appended
              after volume value, the volume is incremented or decremented from the current value, respectively.

```

**Comment by singalsu** [comment]
I checked with UPX set up to IPC4, works there too:

```
$ amixer sset 'gain.1.1 1 2nd' 0dB
Simple mixer control 'gain.1.1 1 2nd',0
  Capabilities: pvolume
  Playback channels: Front Left - Front Right
  Limits: Playback 0 - 45
  Mono:
  Front Left: Playback 45 [100%] [0.00dB]
  Front Right: Playback 45 [100%] [0.00dB]
$ amixer sset 'gain.1.1 1 2nd' 6dB-
Simple mixer control 'gain.1.1 1 2nd',0
  Capabilities: pvolume
  Playback channels: Front Left - Front Right
  Limits: Playback 0 - 45
  Mono:
  Front Left: Playback 42 [93%] [-6.00dB]
  Front Right: Playback 42 [93%] [-6.00dB]

```

**Comment by singalsu** [comment]
> This is still a problem that needs to be clarified.

Maybe there's a difference in topology in working and non-working gains?

**Comment by keqiaozhang** [comment]
> Maybe there's a difference in topology in working and non-working gains?
Not sure about it. Since alsabat test occasionally failed on some IPC4 platforms. So for IPC4 platforms, let's set the gain volume to `100%`. We can switch it back to 0dB after we fixed this issue.

**Comment by marc-hb** [comment]
Fixing this shellcheck warning is not trivial:
https://github.com/thesofproject/sof-test/actions/runs/4423370532/jobs/7756041716
https://www.shellcheck.net/wiki/SC2207
```
 ----- shellcheck -x test-case/volume-basic-test.sh ----


In test-case/volume-basic-test.sh line 61:
pgalist=($(amixer -c"$sofcard" controls | grep -i PGA | sed 's/ /_/g;' | awk -Fname= '{print $2}'))
         ^-- SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
```

Here's one way how. Note `PGA` is now `gain` in IPC4 (thx @fredoh9 !) so I added it.
```shell
ubuntu@sh-mtlp-rvp-hda-03:~$ mapfile pgalist < <(amixer -c"$sofcard" controls |
         awk -Fname= '/PGA|gain/ { gsub(" ","_", $2 ); print $2}')

ubuntu@sh-mtlp-rvp-hda-03:~$ declare -p pgalist
declare -a pgalist=([0]=$'\'gain.1.1_1_2nd_Playback_Volume\'\n' [1]=$'\'gain.11.1_DMIC0_Capture_Volume_1\'\n' [2]=$'\'gain.15.1_Deepbuffer_Volume\'\n' [3]=$'\'gain.2.1_2_Main_Playback_Volume\'\n')
```

**Comment by keqiaozhang** [comment]
> Here's one way how. Note PGA is now gain in IPC4 (thx @fredoh9 !) so I added it.

`volume-basic-test.sh` only support IPC3 ATM, IPC4 is not supported.
In this PR, I only renamed the `reset_PGA_volume` to a generic name `reset_sof_volume`. Will add the volume test for IPC4 platforms later.

**Comment by keqiaozhang** [comment]
CI Test passed on all platforms. Can we merge this PR?  @marc-hb  @singalsu  @fredoh9 
For more tests, Please refer to:https://github.com/thesofproject/sof-test/pull/1012#issuecomment-1473307648

**Comment by marc-hb** [comment]
https://sof-ci.01.org/softestpr/PR1010/build194/devicetest/index.html?model=TGLU_RVP_NOCODEC_IPC4ZPH&testcase=check-alsabat-headset-playback looks like a failure (no idea why)

**Comment by keqiaozhang** [comment]
> https://sof-ci.01.org/softestpr/PR1010/build194/devicetest/index.html?model=TGLU_RVP_NOCODEC_IPC4ZPH&testcase=check-alsabat-headset-playback looks like a failure (no idea why)

It's a known issue on IPC4 nocodec platforms:https://github.com/thesofproject/sof/issues/7285

**Review by miRoox — APPROVED** [review]

**Pull request merged**